src/vis/feature-events.js

```diff
@@ -36,6 +37,13 @@ CartoDBFeatureEvents.prototype._setLayerView = function () {
   }
 };
 
+CartoDBFeatureEvents.prototype._unsetLayerView = function () {
+  if (this._layerView) {
+    this._unbindCartoDBFeatureEvents();
```

Comment:
All CartoDB layers share the same "layerView", since they are grouped together in a single layer. `_unbindCartoDBFeatureEvents` will actually unbinding events from all CartoDB layers, so I'm afraid this will disable events for every other layer, until a new layer is added or layers are reseted :(